src/jwk.rs

```diff
@@ -605,8 +605,12 @@ impl TryFrom<&ECParams> for secp256k1::PublicKey {
         let y = &params.y_coordinate.as_ref().ok_or(Error::MissingPoint)?.0;
         // TODO: add sign byte?
         let pk_data = [x.as_slice(), y.as_slice()].concat();
-        let public_key =
-            secp256k1::PublicKey::parse_slice(&pk_data, Some(secp256k1::PublicKeyFormat::Raw))?;
+        let key_format = match pk_data.len() {
+            64 => Some(secp256k1::PublicKeyFormat::Raw),
+            33 => Some(secp256k1::PublicKeyFormat::Compressed),
+            _ => None,
+        };
+        let public_key = secp256k1::PublicKey::parse_slice(&pk_data, key_format)?;
```

Comment:
As referenced in https://github.com/spruceid/ssi/pull/98/files#r585635887, RFC 8812 (which specifies secp256k1 JWK representation), says:
> As a compressed point encoding representation is not defined for JWK elliptic curve points, the uncompressed point encoding defined there MUST be used.  The "x" and "y" values represented MUST both be exactly 256 bits, with any leading zeros preserved.

So I think the earlier behavior (using `secp256k1::PublicKeyFormat::Raw`, which [requires](https://docs.rs/libsecp256k1/0.3.5/src/secp256k1/lib.rs.html#129-130) the input to be 64 bytes) is correct. We could also check that `x` and `y` are 32 bytes. "TODO: add sign byte?" could probably be removed, since it seems there is no place in the JWK to put the sign byte, and it is reconstructed from the `x` and `y` coordinates anyways.

Reply:
Thanks for the explanations, fixed in 3f8e46f